status/worksets.py

```diff
@@ -59,19 +59,34 @@ def _get_latest_running_note(val):
                     continue
         return result
 
+    def closed_ws_data(self, all=True):
+        result={}
+        a_year_ago = datetime.datetime.now() - relativedelta(years=1)
+        ws_view= self.application.worksets_db.view("worksets/summary", descending=True)
+        for row in ws_view:
+            try:
+                if parse(row.value['date_run']) <= a_year_ago:
+                    for project in row.value['projects']:
+                         if row.value['projects'][project]['status'] == 'Closed':
+                             result[row.key]=row.value
+            except TypeError:
+                continue
+        return result
+
     def get(self):
         # Default is to NOT show all worksets
         all=self.get_argument("all", False)
         t = self.application.loader.load("worksets.html")
         ws_data=self.worksets_data(all=all)
+        closed_worksets_data=self.closed_ws_data(all=all)
```

Comment:
Maybe it would be better to have this as a separate API as it would return all worksets from more than a year old which would keep growing which in turn would keep slowing down the default tab.

Reply:
Valid point! I'm not quite sure how to do that without creating a new handler class, but we can talk about that later :-) 

I also tried adding a time interval just now, so all projects that have been closed for one year until two years are shown in the table, so all projects from 2018 until a year ago:

```
a_year_ago = datetime.datetime.now() - relativedelta(years=1)
two_years_ago = datetime.datetime.now() - relativedelta(years=2)

if parse(row.value['date_run']) <= a_year_ago and parse(row.value['date_run']) >= two_years_ago:
```

I don't know what the lab procedures are for discarding plates, how often they do it. But that would be an alternative if they throw them away within the year. I can talk to the lab!

Reply:
OK, so with help from Anand I transferred some of the code to Javascript in order to load the table separately. Seems like the lab wants all the worksets showing, since they don't discard the plates after a certain amount of time.